### ISSUE TITLE
Hidden Layer Output in pytorch issue

### ISSUE THREAD

**Issue by arjunsoin** [SeldonIO/alibi-detect#241]
Using HiddenOutput for Pytorch, I am constantly running into this issue. Trying to use hidden layer output from a pretrained DenseNet121 for BBSD. Can't seem to figure out if versioning issue or just dimension mismatch (tried that out as well)

`RuntimeError: The following operation failed in the TorchScript interpreter.`

` x2 = x.pow(2).sum(dim=-1, keepdim=True)
    y2 = y.pow(2).sum(dim=-1, keepdim=True)
    dist = torch.addmm(y2.transpose(-2, -1), x, y.transpose(-2, -1), alpha=-2).add_(x2)
           ~~~~~~~~~~~ <--- HERE
    return dist.clamp_min_(a_min)
RuntimeError: mat1 must be a matrix, got 4-D tensor`

**Comment by arnaudvl** [comment]
Hi @arjunsoin , could you provide a reproducible example so I can investigate? E.g. something like:

```python
import torch
from torchvision.models import densenet121
from alibi_detect.cd.pytorch import HiddenOutput

net = densenet121(pretrained=False)
model = HiddenOutput(net, layer=-1)
shape = (1, 3, 64, 64)
x = torch.randn(*shape)
y = model(x)
```

**Comment by arjunsoin** [comment]
Sure, sorry. Here it is: 

```python
import torch
import torch.nn as nn
import torch.optim as optim
from torch.autograd import Variable
import torchvision
from torchvision import datasets, models, transforms

clf = models.densenet121(pretrained=True)
clf.eval()

device = torch.device('cpu')
preprocess_fn = partial(preprocess_drift, model=HiddenOutput(clf, layer=-1),
                        device=device, batch_size=128)

cd = MMDDrift(X_ref_pt, backend='pytorch', preprocess_fn=preprocess_fn, device='cpu') ```

**Comment by arnaudvl** [comment]
Hi @arjunsoin , it looks like the built-in `HiddenOutput` functionality requires additional flattening of its output. The following should resolve your issue:

```python
from typing import Union

class HiddenOutput(nn.Module):
    def __init__(
            self,
            model: Union[nn.Module, nn.Sequential],
            layer: int = -1
    ) -> None:
        super().__init__()
        self.model = nn.Sequential(*list(model.children())[:layer])

    def forward(self, x: torch.Tensor) -> torch.Tensor:
        return nn.Flatten()(self.model(x))
```
I will add this in the next release as well.

**Comment by arnaudvl** [comment]
Addressed in #259
